commit-rights.md

```diff
@@ -0,0 +1,72 @@
+
+This document is based on the [Ansible committer guidelines](https://github.com/ansible/ansible/blob/b57444af14062ec96e0af75fdfc2098c74fe2d9a/docs/docsite/rst/community/committer_guidelines.rst).
+
+These are the guidelines for people with commit privileges on the Ansible Community General Collection GitHub repository. Please read the guidelines before you commit.
+
+These guidelines apply to everyone. At the same time, this is NOT a process document. So just use good judgment. You have been given commit access because we trust your judgment.
+
+That said, use the trust wisely.
+
+If you abuse the trust and break components and builds, and so on, the trust level falls and you may be asked not to commit or you may lose your commit privileges.
+
+Our workflow on GitHub
+----------------------
+
+As a committer, you may already know this, but our workflow forms a lot of our team policies. Please ensure you are aware of the following workflow steps:
+
+* Fork the repository upon which you want to do some work to your own personal repository
+* Work on the specific branch upon which you need to commit
+* Create a Pull Request back to the Ansible repository and tag the people you would like to review; assign someone as the primary "owner" of your request
+* Adjust code as necessary based on the Comments provided
+* Ask someone from the other committers to do a final review and merge
+
+Sometimes, committers merge their own pull requests. This section is a set of guidelines. If you are changing a comma in a doc or making a very minor change, you can use your best judgement. This is another trust thing. The process is critical for any major change, but for little things or getting something done quickly, use your best judgement and make sure people on the team are aware of your work.
+
+Roles
+-----
+* Release managers: Merge pull requests to `stable-X` branches, create tags to do releases.
+* Committers: Fine to do PRs for most things, but we should have a timebox. Hanging PRs may merge on the judgement of these devs.
+* Module maintainers: Module maintainers own specific modules and have indirect commit access through the current module PR mechanisms. This is primary [ansibullbot](https://github.com/ansibullbot)'s `shipit` mechanism.
+
+General rules
+-------------
+Individuals with direct commit access to ansible/ansible are entrusted with powers that allow them to do a broad variety of things--probably more than we can write down. Rather than rules, treat these as general *guidelines*, individuals with this power are expected to use their best judgement.
+
+* Do NOTs:
+
+  - Do not commit directly.
+  - Do not merge your own PRs. Someone else should have a chance to review and approve the PR merge. You have a small amount of leeway here for very minor changes.
+  - Do not forget about alternate environments. Consider the alternatives--yes, people have bad environments, but they are the ones who need us the most.
```

Comment:
I had a hard time understanding this sentence and @felixfontein clarified it for me in #ansible-community IRC channel, but I don't know if there is something that can be added to make it more transparent or if it was just me :)

Reply:
It should definitely be improved then. I'll think about it this evening :)

Reply:
Thanks!

Reply:
@aminvakil what do you think of:
```suggestion
  - Do not forget about non-standard / alternate environments. Consider the alternatives. Yes, people have bad/unusual/strange environments (like binaries from multiple init systems installed), but they are the ones who need us the most.
```

Reply:
I think this clarifies the message, but I'm still not sure if I can understand what do you mean about "binaries from multiple init systems installed" ?

Reply:
Like having `systemctl` installed (from systemd) while you don't actually use systemd. If software like Ansible sees `systemctl` and thinks "this must mean that systemd is the init system", that assumption is wrong for this specific user.

Reply:
That's what I was thinking, but who would have done that? In what purpose?

I'm not sure how many users do this, but is there any other non-standard environments which can be put here as an example?

I would not bother me handling a situation where user has not installed `systemd` but is using `systemctl` really :)

Reply:
I'm not sure if these help or not for custom environments which are more used:

- another shell except bash
- dropping loopback in firewall
- custom linux kernels which do not have common flags enabled during build
- systems without a init by default like alpine

Reply:
> I'm not sure how many users do this, but is there any other non-standard environments which can be put here as an example which is more common?

Apparently it is quite common in containers (for some people at least?).

Reply:
Can't we remove the sentence completely?
The initial one sounds a bit weird to me. Something like: "Don't forget about non-standard strange environments that is freaking hard but can be imagined." or "When merging, try to imagine a weird environment that the changes will break":)
I'd remove it at all...

Reply:
I think we should keep something that describes the basic idea and not remove it completely.

Reply:
Ok, maybe rephrase the initial one generalizing and simplifying

Reply:
> Apparently it is quite common in containers (for some people at least?).

I wasn't aware of this, if that's common then it's fine to me, because as I've said before this clears the message that "beware of weird environments".